### DISCUSSION TITLE
How to get html of selected text

### DISCUSSION THREAD

**Discussion by sakic-milan** [ueberdosis/tiptap#5192] [General Questions]
Hello, I am using tiptap inside Angular app: 

<tiptap-editor [editor]="editor" [(ngModel)]="displayedContent"></tiptap-editor>

I want to be able to get startIndex, endIndex and html string of selected text.

Example:
**HTML:**
<h1>Title</h1>
<p>Test</p>

**Display**:
### Title
Test

If user select "Title te" I need to get "<h1>Title</h1>
<p>Te" and start index 0 and end index 20 (with html tags chars included)

Any ideas?

**Comment by nperez0111** [comment]
```js
import { getHTMLFromFragment } from '@tiptap/core'

editor.chain().focus().command(({ tr }) => {
  console.log(getHTMLFromFragment(tr.doc.slice(tr.selection.from, tr.selection.to).content, editor.schema))
  return true
}).run()
```

Just one way to do it, there are others of course
